jaeger-core/src/main/java/io/jaegertracing/internal/propagation/B3TextMapCodec.java

```diff
@@ -129,6 +140,7 @@ public JaegerSpanContext extract(TextMap carrier) {
 
   public static class Builder {
     private String baggagePrefix = BAGGAGE_PREFIX;
+    private JaegerObjectFactory objectFactory = new JaegerObjectFactory();
```

Comment:
You could use the service loader API here, so that having a JaegerObjectFactory in the classpath would be sufficient to use an alternate object factory. Look at how we did it for the MetricsFactory/Sender for a reference.

Reply:
I had actually been considering something along those lines, but didn't realize there were other implementations. Thanks!